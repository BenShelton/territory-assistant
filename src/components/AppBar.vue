<template>
  <v-app-bar app>
    <v-app-bar-nav-icon @click.stop="toggleDrawer" />
    <v-toolbar-title class="headline">
      <span>Territory Assistant</span>
    </v-toolbar-title>
  </v-app-bar>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'AppBar',

  methods: {
    toggleDrawer (): void {
      this.$store.commit('drawer/toggleDrawer')
    }
  }
})
</script>
