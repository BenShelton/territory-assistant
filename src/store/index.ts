import Vue from 'vue'
import Vuex from 'vuex'
import createPersistedState from 'vuex-persistedstate'

import auth from './modules/auth'
import drawer from './modules/drawer'
import info from './modules/info'
import map from './modules/map'
import notification from './modules/notification'
import settings from './modules/settings'

import { IRootState } from 'types/vuex'

Vue.use(Vuex)

export default new Vuex.Store<IRootState>({
  modules: {
    auth,
    drawer,
    info,
    map,
    notification,
    settings
  },
  plugins: [createPersistedState({
    paths: ['auth']
  })]
})
