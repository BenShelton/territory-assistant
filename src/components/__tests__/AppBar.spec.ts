import { mount } from '@vue/test-utils'

import AppBar from '../AppBar.vue'
import store from '@/store'

describe('Components/AppBar', () => {
  it('should have the name "AppBar"', () => {
    const wrapper = mount(AppBar, { store })
    expect(wrapper.name()).toBe('AppBar')
  })

  it('should match the snapshot', () => {
    const wrapper = mount(AppBar, { store })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
