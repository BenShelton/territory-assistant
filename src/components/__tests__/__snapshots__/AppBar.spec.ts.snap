// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/AppBar should match the snapshot 1`] = `
<header class="v-sheet v-sheet--tile theme--light v-toolbar v-app-bar v-app-bar--fixed" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
  <div class="v-toolbar__content" style="height: 64px;">
    <!---->
    <div class="v-toolbar__title headline"><span>Territory Assistant</span></div>
  </div>
</header>
`;
